readme.md

```diff
@@ -1,50 +1,79 @@
-# vdom-streaming-serializer
-
 [![Build Status](https://travis-ci.org/donejs/vdom-streaming-serializer.png?branch=master)](https://travis-ci.org/donejs/vdom-streaming-serializer)
 
+# vdom-streaming-serializer
 
+The vdom-streaming-serializer is a library for serializing virtual DOM objects into chunks of strings that can be written into an HTTP response.
 
-## Usage
-
-### ES6 use
+## Installation
+```
+npm install vdom-streaming-serializer
+```
 
-With StealJS, you can import this module directly in a template that is autorendered:
+## Usage
+A virtual dom has to be created and running the serialization will emit streams of asynchronous elements.
 
+### Example
+This example creates a virtual dom and serializes it to emit streams of html elements in Chunks.
 ```js
-import plugin from 'vdom-streaming-serializer';
+var document = makeDocument();
+var ul = document.createElement('ul');
+document.body.appendChild(ul);
+var li = document.createElement('li');
+```
+Making this ```html <li> ``` as asynchronous will force the serialize to wait.
+```js
+var ASYNC = Symbol.for('async-node');
+li[ASYNC] = Promise.resolve();
+ul.appendChild(li);
 ```
 
-### CommonJS use
-
-Use `require` to load `vdom-streaming-serializer` and everything else
-needed to create a template that uses `vdom-streaming-serializer`:
-
+Once the element is serialized, it is then emitted in ```stream.on```.
 ```js
-var plugin = require("vdom-streaming-serializer");
+var stream = serialize(document);
+stream.setEncoding('utf8');
+stream.on('data', function(html){
+	console.log('Chunk', html);	
+});
+```
+The output from this demo should be:
 ```
+Chunk <html><body><ul>
+Chunk <li></li></ul<body></html>
+```
+From this, the ```html <li> ``` element is serialized asynchronously.
 
-## AMD use
 
-Configure the `can` and `jquery` paths and the `vdom-streaming-serializer` package:
+## API
+The serialization process traverses and checks for html syntax and attributes to add to the buffer.
+```js
```

Comment:
I would remove this code snippet as it is internal code.  The explanation is great though!